### PR TITLE
Enable logging and proper initialize the session

### PR DESCRIPTION
**Pull request by NonStatic2014**
Sample of logging:

```
/home/klein/code/onnxruntime/cmake/build/CLion/onnxruntime_hosting -a 0.0.0.0 -p 8901 -t 10 -m /home/klein/code/onnx/models/mnist/mnist/model.onnx
2019-03-14 10:14:36.430724047 [V:onnxruntime:HostingLog, main.cc:66 main] Logging manager initialized.
2019-03-14 10:14:36.430768484 [V:onnxruntime:HostingLog, main.cc:67 main] Model path: /home/klein/code/onnx/models/mnist/mnist/model.onnx
2019-03-14 10:14:36.432149181 [V:onnxruntime:HostingLog, main.cc:69 main] Load Model Status: 0 ---- Error: []
Listening at: 
 
    http://0.0.0.0:8901
```

### CODE REVIEW

File: onnxruntime/hosting/environment.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "log_sink.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+class HostingEnvironment {
+ public:
+  HostingEnvironment();
+
+  const onnxruntime::logging::Logger& GetLogger();
```
Comment (tmccrmck) [inline onnxruntime/hosting/environment.h:21]: Google style guide wants const after `GetLogger()` 

https://google.github.io/styleguide/cppguide.html#Use_of_const

File: onnxruntime/hosting/environment.cc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                           default_logging_manager_(
+                                               std::unique_ptr<onnxruntime::logging::ISink>{&sink_},
+                                               onnxruntime::logging::Severity::kVERBOSE,
+                                               /* default_filter_user_data */ false,
+                                               onnxruntime::logging::LoggingManager::InstanceType::Default,
+                                               &logger_id_) {
+  auto status = onnxruntime::Environment::Create(this->runtime_environment_);
+
+  // The session initialization MUST BE AFTER environment creation
+  session_ = std::make_shared<onnxruntime::InferenceSession>(options_, &default_logging_manager_);
```
Comment (tmccrmck) [inline onnxruntime/hosting/environment.cc:23]: Question: why does this need to be shared?

**Review by tmccrmck — APPROVED** [review]

File: onnxruntime/hosting/environment.h (hunk 1 above)
Comment (NonStatic2014) [reply onnxruntime/hosting/environment.h:21]: For this case, the return value is `const onnxruntime::logging::Logger&`. So I will leave it as-is. But the `GetSessin()` should be a `const` method. I will update it.

File: onnxruntime/hosting/environment.cc (hunk 2 above)
Comment (NonStatic2014) [reply onnxruntime/hosting/environment.cc:23]: The `session_` will be shared accross all worker threads.

**Pull request merged**